inc/3rdparty/diff/Diff/Engine/Native.php

```diff
@@ -196,18 +196,30 @@ protected function _diag ($xoff, $xlim, $yoff, $ylim, $nchunks)
                     continue;
                 }
                 $matches = $ymatches[$line];
-                foreach ($matches as $y) {
+                reset($matches);
+                while ($y = current($matches)) {
```

Comment:
Reading [the docs for `current`](http://php.net/manual/en/function.current.php) and the return value, shouldn't we explicitly check for `$y !== false` here?

Reply:
Indeed; cc @johnbillion @SergeyBiryukov